src/loaders/gltf.rs

```diff
@@ -13,31 +15,97 @@ pub fn load_file(path: impl AsRef<Path>) -> Result<Model, gltf::Error> {
 pub struct GltfFile {
     document: gltf::Document,
     buffers: Vec<gltf::buffer::Data>,
+    materials: Vec<Arc<Material>>,
+}
+
+/// glTF node that can be transformed
+struct Node<'a> {
+    gltf_node: gltf::Node<'a>,
+    translation: Vec3<f32>,
+    rotation: Quaternion<f32>,
+    scale: Vec3<f32>,
+}
+
+impl<'a> From<gltf::Node<'a>> for Node<'a> {
+    fn from(gltf_node: gltf::Node<'a>) -> Self {
+        Node {
+            gltf_node,
+            translation: Vec3::zero(),
```

Comment:
I (probably) eventually want to store all of all these together in one transformation matrix. We can talk about that more on Friday though.